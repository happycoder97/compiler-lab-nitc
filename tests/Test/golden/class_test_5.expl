class A         
{
  decl
    int f0();
    int f1();
  enddecl
  int f0() {               
      begin
       write("In class A f0");
       return 1;
      end
  }
  int f1() {                  
      begin
        write("In class A f1");
        return 1;
      end
  }
}                      
class B extends A
{
  decl
    int f0();
    int f2();
  enddecl
int f0() {                    
     begin
       write("In class B f0");
       return 1;
     end
}
int f2() {                  
     begin
       write("In class B f2");
       return 1;
     end
}


}                      
class C extends B
{
  decl
    int f0();
    int f2();
    int f4();
  enddecl
int f0() {                
     begin
       write("In class C f0");
       return 1;
     end
}
int f2() {                
    begin
       write("In class C f2");
       return 1;
    end
}
int f4() {              
    begin
       write("In class C f4");
       return 1;
    end
}
}                       

decl
    A obj ;
    A test_obj;
enddecl

int main() {
  decl
    int temp,n;
  enddecl
  begin
    initialize();
    read(n);
    if(n < 0)then
        obj = new(A);
    else
        if(n == 0)then
            obj = new(B);
        else
            if(n > 0)then
                obj = new(C);
            endif;
       endif;
    endif;
   test_obj = obj;
   write(test_obj.f0());              
   return 1;
  end
}
