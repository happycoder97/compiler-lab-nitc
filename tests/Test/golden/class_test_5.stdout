In class A f0
1
